src/components/PostDetail/components/PostDetailInfo/components/PostDetailComment/components/PostCommentInput/PostCommentInput.tsx

```diff
@@ -1,7 +1,135 @@
+import { ChangeEvent, FormEvent, MouseEvent, useState } from "react"
+import ConfirmModal from "@/components/Modal/components/ConfirmModal/ConfirmModal"
 import * as S from "./PostCommentInput.Styles"
+import ArrowUpwardIcon from "@mui/icons-material/ArrowUpward"
+import useModal from "@/components/Modal/hooks/useModal"
+import { POST_DETAIL_MODAL_MESSAGE } from "@/constants/modalMessage"
+import { Post } from "@/types"
+import useAuthUserStore from "@/stores/useAuthUserStore"
+import { useNavigate } from "react-router-dom"
+import { useMutation, useQueryClient } from "@tanstack/react-query"
+import { sendComment } from "@/components/PostDetail/apis/sendComment"
+import commentValidation from "./util/commentValidation"
+import { POST_DETAIL_ERROR_MESSAGE } from "@/constants/errorMessage"
+import AlertModal from "@/components/Modal/components/AlertModal/AlertModal"
 
-const PostCommentInput = () => {
-  return <S.PostCommentInputLayout></S.PostCommentInputLayout>
+const SEND_MUTATION_QUERY_KEY = "SEND_MUTATION_MUTATION_KEY_9128178621782"
+
+interface PostCommentInputProps {
+  post: Post
+}
+
+const PostCommentInput = ({ post }: PostCommentInputProps) => {
+  const [writeComment, setWriteComment] = useState("")
+  const [alertMessage, setAlertMessage] = useState("")
+  const { user: authUser, isLoggedIn } = useAuthUserStore()
+
+  const {
+    isShowModal: isShowConfirm,
+    showModal: showConfirm,
+    closeModal: closeConfirm,
+  } = useModal()
+  const {
+    isShowModal: isShowAlert,
+    showModal: showAlert,
+    closeModal: closeAlert,
+  } = useModal()
+
+  const queryClient = useQueryClient()
+  const CommentApi_send = useMutation({
+    mutationKey: [SEND_MUTATION_QUERY_KEY],
+    mutationFn: sendComment,
+    onSuccess: () => {
+      queryClient.refetchQueries()
+      setWriteComment("")
+    },
+    onError: () => {
+      setAlertMessage(POST_DETAIL_ERROR_MESSAGE.COMMENT_SUBMIT.ERROR)
+      showAlert()
+    },
+  })
+
+  const navigate = useNavigate()
+
+  const handleClickLoginCheck = (e: MouseEvent) => {
+    e.preventDefault()
+    if (isLoggedIn) {
+      return
+    }
+    showConfirm()
+  }
+
+  const handleCloseConfirmModal = (isAccept: boolean) => {
+    if (isAccept) {
+      closeConfirm()
+      navigate("/login")
+      return
+    }
+    closeConfirm()
```

Comment:
[P999] 가독성 때문에 이렇게 하신 것 같긴 한데 confirm 모달창을 무조건 닫기 때문에 이렇게도 바꿔 볼 수 있을 것 같아요!! 🤔
```suggestion
    closeConfirm()
    if (isAccept) {
      navigate("/login")
    }
```

Reply:
오? 그렇네요!  약간 얼리 리턴이 익숙해지니 이런걸 놓치네요

Reply:
적용 완료